content/en/serverless/google_cloud_run/_index.md

```diff
@@ -2,201 +2,296 @@
 ENTRYPOINT ["/app/datadog-init"]
-CMD ["dotnet", "helloworld.dll"] (adapt this line to your needs)
 
-```
+# execute your binary application wrapped in the entrypoint. Adapt this line to your needs
+CMD ["dotnet", "helloworld.dll"]
 
-See [Tracing .NET Applications][1] for detailed instructions. [Sample code for a simple .NET application][2].
+```
 
-[1]: /tracing/trace_collection/dd_libraries/dotnet-core?tab=containers
-[2]: https://github.com/DataDog/crpb/tree/main/dotnet
 {{< /programming-lang >}}
 {{< programming-lang lang="ruby" >}}
+
 ```
-COPY --from=datadog/serverless-init:beta4 /datadog-init /app/datadog-init
+# copy the Datadog `serverless-init` into your Docker image
+COPY --from=datadog/serverless-init /datadog-init /app/datadog-init
+
+# optionally add Datadog tags
+ENV DD_SERVICE=datadog-demo-run-ruby
+ENV DD_ENV=datadog-demo
+ENV DD_VERSION=1
+
+# change the entrypoint to wrap your application into the Datadog serverless-init process
 ENTRYPOINT ["/app/datadog-init"]
+
+# execute your binary application wrapped in the entrypoint. Adapt this line to your needs
 CMD ["rails", "server", "-b", "0.0.0.0"] (adapt this line to your needs)
 
 ```
 
-See [Tracing Ruby Applications][1] for detailed instructions. [Sample code for a simple Ruby application][2].
+[Sample code for a simple Ruby application][1].
+More details on the [Ruby tracing library][2].
+
+[1]: https://github.com/DataDog/crpb/tree/main/ruby-on-rails
+[2]: /tracing/trace_collection/dd_libraries/ruby/?tab=containers#instrument-your-application
 
-[1]: /tracing/trace_collection/dd_libraries/ruby/
-[2]: https://github.com/DataDog/crpb/tree/main/ruby-on-rails
 {{< /programming-lang >}}
 {{< /programming-lang-wrapper >}}
 
-#### Troubleshooting
-This integration depends on your runtime having a full SSL implementation. If you are using a slim image for Node, you may need to add the following command to your Dockerfile to include certificates.
+#### 1.b Instrument using buildpack
 
-```
-RUN apt-get update && apt-get install -y ca-certificates
-```
+[`Pack Buildpacks`][4] provide a convenient way to package your container without using a Dockerfile. This example uses the GCP container registry and Datadog serverless buildpack.
 
-#### Build with the Datadog buildpack
+**Note**: Follow the instructions to install the [tracing library][2] for your language before running the buildpack.
 
-1. Use [`pack`][4] to build your application by running the following:
-   ```
+Build your application by running the following command:
+
+   ```shell
    pack build --builder=gcr.io/buildpacks/builder \
    --buildpack from=builder \
-   --buildpack datadog/serverless-buildpack:beta4 \
+   --buildpack datadog/serverless-buildpack \
    gcr.io/YOUR_PROJECT/YOUR_APP_NAME
-
    ```
 
-   **Note**: Not compatible with Alpine.
+**Note**: Not compatible with Alpine.
 
- 2. Push your image to GCP:
-    ```
-    docker push gcr.io/YOUR_PROJECT/YOUR_APP_NAME
-    ```
+### 2. Datadog Agent configuration
```

Comment:
"Configure your application"? or "Configure the instrumentation"? it's not only about configuring the agent (but also tracer, etc)